flink-runtime/src/main/java/org/apache/flink/runtime/security/token/DefaultDelegationTokenManager.java

```diff
@@ -213,6 +213,20 @@ public void obtainDelegationTokens(DelegationTokenContainer container) throws Ex
         LOG.info("Delegation tokens obtained successfully");
     }
 
+    @Override
+    public void obtainDelegationTokens() throws Exception {
```

Comment:
We need to the immediately propagate the existing tokens (fetched here) to Task managers once start function is called. Otherwise, I see No credential error in task manager. I guess it is due to the latency of additional token fetching in startTokenUpdate function.

Reply:
At the one-time token obtain stage there are no task managers but I see that there is an issue so I'll take a look...

Reply:
I've just added further changes to solve the TM issue. We should test it on cluster in-depth because that's modifying Flink's critical path.

Reply:
@gaborgsomogyi  Sorry for late rely. The feature is fully tested end to end.